### PR TITLE
Work around incorrect content-disposition headers in some VSO files

### LINKED ISSUE #3372: Mangled filename when downloading MDI data from VSO using Fido
### Description

When downloading MDI data from VSO the files are saved with incorrect filenames, parts of the content-encoding header are being included in the file name. It looks like the headers are not being set correctly on the response sent from VSO but this may not be easily fixed so a work around may be required.

### Expected behavior

Files to be saved with the correct filename

### Actual behavior

Files are not save with the correct filename

### Steps to Reproduce

```python
results = Fido.search(a.Time('2011/1/1 01:00', '2011/1/1 01:02'), a.Instrument(hmii') | a.Instrument('mdi'))
files = Fido.fetch(results[:,0])
print(files)
['/Users/shane/sunpy/data/hmi_ic_45s_2011_01_01_01_01_30_tai_continuum.fits', '/Users/shane/sunpy/data/mdi_vw_v_9466622_9466622.tar"\\nContent-transfer-encoding: binary']
```
### System Details

  - SunPy Version: 1.0.2
  - Astropy Version: 3.2.1 
  - Python Version: 3.7.0
  - OS information: macOS 10.14.6

### PR DESCRIPTION
**Pull request by Cadair**
Fixes #3372

### CODE REVIEW

File: sunpy/net/vso/vso.py [hunk 1]
```diff
@@ -414,26 +416,45 @@ def merge(self, queryresponses):
 
     @staticmethod
     def mk_filename(pattern, queryresponse, resp, url):
+        """
+        Generate the best possible (or least-worse) filename for a VSO download.
+
+        * Use the ``content-disposition`` header.
+        * Use `providerid_fileid` to generate a unique file ID if content-disposition fails
```
Comment (Cadair) [inline sunpy/net/vso/vso.py]:
```suggestion
        * Use `fileid` to generate a file name if content-disposition fails
```

File: sunpy/net/tests/test_vso.py [hunk 2]
```diff
@@ -475,3 +475,15 @@ def test_vso_error(client):
...
+
+@pytest.mark.remote_data
+def test_incorrect_content_disposition(client):
+    results = client.search(
+        va.Time('2011/1/1 01:00', '2011/1/1 01:02'),
+        va.Instrument('mdi'))
+    files = client.fetch(results[0:1])
+
+    assert len(files) == 1
+    assert  files[0].endswith("mdi_vw_v_9466622_9466622.tar")
```
Comment (samaloney) [inline sunpy/net/tests/test_vso.py]: Not necessarily an issue with this PR but that file looks wrong (the name and tar extension). When I download it and expand the tar it contains a file called `error_list.txt` and this has the following entry
```
record = mdi.vw_V[2011.01.01_01:02:00_TAI], file = mdi.vw_V.2011.01.01_01:02:00_TAI.data.tas, message = cannot export Rice-compressed floating-point images
```

Comment (Cadair) [reply sunpy/net/tests/test_vso.py]: I have opened https://github.com/sunpy/sunpy/issues/3747 to track this and have asked the VSO team about it.

**Comment by Cadair** [comment]
The HEK seems to be having issues, given the last commit is a typo in a comment, and the one before was green I am going to merge this.

**Pull request merged**